backend/src/packages/users/user.repository.ts

```diff
@@ -32,7 +39,13 @@ class UserRepository implements IRepository {
       .returning('*')
       .execute();
 
-    return UserEntity.initialize(item);
+    await this.userDetailsModel.query().insert({
+      firstName,
+      lastName,
+      userId: item.id,
+    });
+
+    return UserEntity.initialize({ ...item, firstName, lastName });
```

Comment:
Then if you'd use `insertGraphAndFetch` for insertion - you may return entity like this
```suggestion
    return UserEntity.initialize({
      id: user.id,
      email: user.email,
      passwordHash: user.passwordHash,
      passwordSalt: user.passwordSalt,
      firstName: user.userDetails.firstName,
      lastName: user.userDetails.lastName
    });
```